node/src/components/block_accumulator/block_acceptor.rs

```diff
@@ -129,7 +130,29 @@ impl BlockAcceptor {
         &mut self,
         finality_signature: FinalitySignature,
         peer: Option<NodeId>,
+        validator_slots: u32,
     ) -> Result<Option<FinalitySignature>, AcceptorError> {
+        /// Returns an error if the peer has sent too many finality signatures.
+        fn check_signatures_from_peer_bound(
+            limit: u32,
+            peer: NodeId,
+            signatures: &BTreeMap<PublicKey, (FinalitySignature, BTreeSet<NodeId>)>,
+        ) -> Result<(), AcceptorError> {
+            let signatures_for_peer = signatures
+                .values()
+                .filter(|(_fin_sig, nodes)| nodes.contains(&peer))
+                .count();
```

Comment:
There is a potential performance impact with iterating through this collection for each signature being registered, but I think that with the sizes of the collections we're dealing with here that should not be a problem.

Let's see what others say.

Reply:
Is there any other way to do this?

Reply:
So far there's no need to change the implementation.